crates/distribution-types/src/lib.rs

```diff
@@ -424,11 +434,28 @@ impl Dist {
     }
 }
 
+impl From<RegistryBuiltWheel> for Dist {
+    fn from(wheel: RegistryBuiltWheel) -> Dist {
+        let regdist = RegistryBuiltDist {
+            wheels: vec![wheel],
+            best_wheel_index: 0,
+            sdist: None,
+        };
+        Dist::Built(BuiltDist::Registry(regdist))
+    }
+}
```

Comment:
Where is this used? It seems somewhat lossy, is it dangerous that it exists?

Reply:
Yeah this is a really nice catch. I had added this to help with the refactor, but it turned out it was still being used to convert a `ResolvedDistRef` into a `Request` in the resolver. This was indeed dropping all of the wheels that were being carried around, although I don't know much it matters. (It's also another copy of the dist data.)

I've removed this `From` impl and added a `From<ResolvedDistRef> for Request` impl that does the conversion correctly.